packages/atlas/src/providers/transactionManager/useTransaction.ts

```diff
@@ -111,35 +113,43 @@ export const useTransaction = (): HandleTransactionFn => {
         await queryNodeSyncPromise
 
         if (result.transactionHash) {
-          const { data } = await getTransactionStatus({ variables: { transactionHash: result.transactionHash } })
+          await getTransactionStatus({ variables: { transactionHash: result.transactionHash } })
           const status = data?.metaprotocolTransactionStatusEvents[0]?.status
 
+          if (status?.__typename === 'MetaprotocolTransactionErrored') {
+            throw new JoystreamLibError({
+              name: 'MetaprotocolTransactionError',
+              message: status?.message,
+              details: result,
+            })
+          }
+
           if (!status) {
             for (let i = 0; i <= RETRIES; i++) {
-              ConsoleLogger.warn(`No transcation status event found - retries: ${i}/${RETRIES}`)
+              ConsoleLogger.warn(`No transaction status event found - retries: ${i}/${RETRIES}`)
               await wait(RETRIES_TIMEOUT)
-              const { data } = await refetch()
-              const status = data?.metaprotocolTransactionStatusEvents[0]?.status
-              if (status) {
-                break
+              const { data: refetchedData } = await refetchTransactionStatus()
+              const statusAfterRefetch = refetchedData?.metaprotocolTransactionStatusEvents[0]?.status
+
+              if (statusAfterRefetch?.__typename === 'MetaprotocolTransactionErrored') {
+                throw new JoystreamLibError({
+                  name: 'MetaprotocolTransactionError',
+                  message: statusAfterRefetch?.message,
+                  details: result,
+                })
               }
```

Comment:
I don't think we need to duplicate this. You could just change the declaration in L117 to `let status = ...` and later just override `status` instead of creating new variable. Then after the retry block you can do a single `if` catching both cases